tests/test_pebble_protocol.py

```
@@ -0,0 +1,535 @@
+    assert errors == []
+
+
+def test_libpebble2_send_discards_stale_result_for_reused_transaction() -> None:
+    backend, appmessage = _libpebble_backend()
+    backend._delivery_results[7] = True
+    errors: list[BaseException] = []
+
+    def send() -> None:
+        try:
+            backend.send({})
+        except BaseException as error:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)